stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,477 @@
+                account = categ.property_account_expense_categ_id
+        account = move._get_account(fiscal_position, account)
+        quantity = 0
+        for move in moves:
+            qty = move.product_uom_qty
+            loc = move.location_id
+            loc_dst = move.location_dest_id
+            # Better to understand with IF/ELIF than many OR
+            if inv_type == 'out_invoice' and loc.usage == 'customer':
+                qty *= -1
+            elif inv_type == 'out_refund' and loc_dst.usage == 'customer':
+                qty *= -1
+            elif inv_type == 'in_invoice' and loc_dst.usage == 'supplier':
+                qty *= -1
+            elif inv_type == 'in_refund' and loc.usage == 'supplier':
+                qty *= -1
+            quantity += qty
+        taxes = moves._get_taxes(fiscal_position)
+        price = moves._get_price_unit_invoice(
+            inv_type, invoice.partner_id, quantity)
+        line_obj = self.env['account.invoice.line']
+        values = line_obj.default_get(line_obj.fields_get().keys())
+        values.update({
+            'name': name,
+            'account_id': account.id,
+            'product_id': product.id,
+            'uom_id': product.uom_id.id,
+            'quantity': quantity,
+            'price_unit': price,
+            'invoice_line_tax_ids': [(6, 0, taxes.ids)],
+            'move_line_ids': [(4, m.id, False) for m in moves],
+        })
+        values = self._simulate_invoice_line_onchange(values)
+        return values
+
+    @api.multi
+    def _update_picking_invoice_status(self, pickings):
+        """
+        Update invoice_state on pickings
+        :param pickings: stock.picking recordset
+        :return: stock.picking recordset
+        """
+        return pickings._set_as_invoiced()
+
+    @api.multi
+    def _action_generate_invoices(self):
+        """
+        Action to generate invoices based on pickings
+        :return: account.invoice recordset
+        """
+        pickings = self._load_pickings()
+        company = pickings.mapped("company_id")
+        if company and company != self.env.user.company_id:
+            raise UserError(_("All pickings are not related to your company!"))
+        pick_list = self._group_pickings(pickings)
+        invoices = self.env['account.invoice'].browse()
+        for pickings in pick_list:
+            invoice_values = self._build_invoice_values_from_pickings(pickings)
+            invoice = self.env['account.invoice'].create(invoice_values)
+            moves = pickings.mapped("move_lines")
+            moves_list = self._group_moves(moves)
+            lines = []
+            for moves in moves_list:
+                line_values = self._get_invoice_line_values(moves, invoice)
+                if line_values:
+                    lines.append(line_values)
+            if lines:
+                invoice.write({
+                    'invoice_line_ids': [(0, False, l) for l in lines],
```

Comment:
try to change:

lines = [] to  lines = [(5, 0, {})] 

and 

lines.append(line_values) to append((0, 0, line_values))

Reply:
and 

                    'invoice_line_ids': [(0, False, l) for l in lines],
to 

 invoice.write({ 'invoice_line_ids': lines})